### PR TITLE
T1298

### PR DESCRIPTION
**Pull request by secretrobotron**

### CODE REVIEW

File: src/dependencies.js [hunk 1]
```diff
@@ -69,16 +74,65 @@ define([], function(){
...
+        // Load less.js so we can parse the *.less -> *.css
+        _loaders.js( config.value( "dirs" ).tools + LESS, exclude,
+          function onLoadCallback(){
+            // Assume *.less is beside *.css file
+            var less = window.less,
+                lessFile;
+
+            // Load the .less file, parse with LESS, and inject CSS <style>
+            XHR.get( url, function xhrCallback(){
+              if ( this.readyState === 4) {
```
Comment (mjschranz) [inline src/dependencies.js]: Space after the 4

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/6ca94201f20300b/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/6ca94201f20300b/output.txt

Total script time: 0.40 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/cbc8f5ded44bc50/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/cbc8f5ded44bc50/output.txt

Total script time: 0.60 mins

Server is running on http://pr638.butyr.org/templates/test.html

File: cornfield/app.js [hunk 2]
```diff
@@ -5,11 +5,12 @@ const express = require('express'),
       path = require('path'),
       app = express.createServer(),
       MongoStore = require('connect-mongo')(express),
-      stylus = require('stylus'),
+      lessMiddleware = require('less-middleware'),
       CONFIG = require('config'),
       TEMPLATES_DIR =  CONFIG.dirs.templates,
       PUBLISH_DIR = CONFIG.dirs.publish,
       PUBLISH_PREFIX = CONFIG.dirs.hostname,
+      CSS_DIR = CONFIG.dirs.css,
```
Comment (jbuck) [inline cornfield/app.js]: This looks like it isn't being used for anything, can this be removed?

File: cornfield/config/default.json [hunk 3]
```diff
@@ -16,6 +16,7 @@
   "dirs": {
     "wwwRoot": "../",
     "templates": "../templates",
+    "css": "../css",
```
Comment (jbuck) [inline cornfield/config/default.json]: Same as the note in cornfield/app.js, if this variable isn't being used for anything then nuke it

File: make.js [hunk 4]
```diff
@@ -1,21 +1,40 @@
 #!/usr/bin/env node
 
-var JSLINT = './node_modules/jshint/bin/hint',
-    CSSLINT = './node_modules/csslint/cli.js',
-    UGLIFY = './node_modules/uglify-js/bin/uglifyjs',
-    RJS    = './node_modules/requirejs/bin/r.js',
-    STYLUS = './node_modules/stylus/bin/stylus',
-    DOX    = './tools/dox.py',
-    DIST_DIR = 'dist',
+var path = require( "path"),
```
Comment (jbuck) [inline make.js]: spacing beside the right bracket

File: make.js [hunk 5]
```diff
@@ -158,8 +200,10 @@ function build( version ){
   sed('-i', '@VERSION@', version, 'dist/butter.js');
   sed('-i', '@VERSION@', version, 'dist/butter.min.js');
 
-  exec(STYLUS + ' css');
-  cp('css/*.css', DIST_DIR);
+  // Compress CSS for deployment
+  lessToCSS( true );
+
+  cp( normalize( 'css/*.css' ), DIST_DIR);
```
Comment (jbuck) [inline make.js]: Change this to `cp( BUTTER_CSS_FILE, DIST_DIR);`

**Pull request merged**